plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlConnectorTest.java

```diff
@@ -283,17 +281,10 @@ public void testPredicatePushdown()
                 .matches("VALUES " +
                         "(BIGINT '3', BIGINT '19', CAST('ROMANIA' AS varchar(25))), " +
                         "(BIGINT '2', BIGINT '21', CAST('VIETNAM' AS varchar(25)))")
-                // Filter node is retained as no constraint is pushed into connector.
-                // The compacted domain is a range predicate which can give wrong results
-                // if pushed down as PostgreSQL has different sort ordering for letters from Trino
                 .isNotFullyPushedDown(
                         node(
                                 FilterNode.class,
-                                // verify that no constraint is applied by the connector
-                                tableScan(
-                                        tableHandle -> ((JdbcTableHandle) tableHandle).getConstraint().isAll(),
```

Comment:
retain TS.contraint assertion.

Reply:
Just checking the existence of `FilterNode` is not enough as other many test cases that check whether predicate is not fully pushed down?

Reply:
I think this test included the TS.constraint because it was verifying that the connector didn't consume the filters which isn't the case anymore after this change (because the IN gets rewritten to a range predicate which now gets pushed down to the connector).

@takezoe If you revert this test change I assume the test fails?

Reply:
Oh nevermind, I misread the test. The "actual" predicate would still remain on the engine's side but the range predicate would get pushed down to connector.

Let's change the assertion to verify both assumptions - the tableScan has a range constraint applied and there is a filter node with the actual predicates applied.

Reply:
> Oh nevermind, I misread the test. The "actual" predicate would still remain on the engine's side but the range predicate would get pushed down to connector.

Yes, that's right.

> Let's change the assertion to verify both assumptions - the tableScan has a range constraint applied and there is a filter node above the tableScan.

Okay, let me try.

Reply:
Fixed the test case.

Reply:
Can you add a comment about why this assertion is testing in depth? I think we'll forget eventually.
Something like `Verify that a FilterNode is retained and only a compacted domain is pushed down to connector as a range predicate`.

Reply:
Added a comment.